### PR TITLE
MGDOBR-843 Fix spectral lint errors for rhose-api.yaml

### PR DESCRIPTION
**Pull request by jomarko**
## Dependent on
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/894

For more details see:
- https://issues.redhat.com/browse/MGDOBR-843

[Insert JIRA Issue number and update link](https://issues.redhat.com/browse/MGDOBR-)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/models/responses/ErrorResponse.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(name = "ErrorResponse",
+        description = "General error response",
+        allOf = { ObjectReference.class, ErrorResponse.class })
+
+public class ErrorResponse extends ObjectReference {
```
Comment (danielezonca) [inline manager/src/main/java/com/redhat/service/smartevents/manager/api/models/responses/ErrorResponse.java:12]: Isn't this the same of [`ErrorResponse`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java)? In general we should avoid to have more classes with same name because the openAPI renames it as `ErrorResponse1`

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/models/responses/ObjectReference.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package com.redhat.service.smartevents.manager.api.models.responses;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+public class ObjectReference {
```
Comment (danielezonca) [inline manager/src/main/java/com/redhat/service/smartevents/manager/api/models/responses/ObjectReference.java:6]: Same comment for this and [`BaseResponse`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/BaseResponse.java)

**Comment by jomarko** [comment]
@danielezonca thank you for good pointers. Trying to use existing classes now instead of adding new classes.

**Comment by dimakis** [comment]
@jomarko I've had a quick look through the PR in particular the errors and warnings. 
They can be fixed easily enough, I've outlined how here https://docs.google.com/document/d/15TtXnlHVWmifEli6jVwbn68HwACS9ruXNRBz4y-L2XU/edit?usp=sharing 
any questions just ping me 👍

**Comment by jomarko** [comment]
@dimakis thank you for a document. Could you please grant me an access to it?

**Review by danielezonca — COMMENTED** [review]
@jomarko 
You can update the [`openapi_validate`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/.github/workflows/openapi_validate.yaml#L5) to `pull_request` temporarily in the PR to verify if the issues are solved and then revert it.
Wdyt?

**Comment by jomarko** [comment]
@dimakis @wtrocki hello, I pushed as much fixes for `npx @stoplight/spectral@5.9.0 lint ./openapi/rhose-api.yaml`, however I struggle to understand and fix:
```
  977:11    error  rhoas-error-schema             "type" should be a "object", got "undefined"                                                    components.schemas.Error
 1014:10    error  rhoas-list-schema              "type" should be a "object", got "undefined"                                                    components.schemas.List
```

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java [hunk 3]
```diff
@@ -119,11 +120,11 @@ public Response addProcessorToBridge(@NotEmpty @PathParam("bridgeId") String bri
     @APIResponses(value = {
             @APIResponse(description = "Accepted.", responseCode = "202",
                     content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = ProcessorResponse.class))),
-            @APIResponse(description = "Bad request.", responseCode = "400", content = @Content(mediaType = MediaType.APPLICATION_JSON)),
+            @APIResponse(description = "Bad request.", responseCode = "400", content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = ErrorResponse.class))),
```
Comment (manstis) [inline manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java:123]: My fear is that we return *either* `ErrorResponse` or `ErrorsResponse`.

We need support for multiple errors and hence, if limited to one response type, we should move all of RHOSE to return `ErrorsResponse` for consistency; but that puts a burden on Users and most certainly our UI development.

The same can be said for `BridgesAPI`; its methods can return multiple errors (i.e `ErrorsResponse`).

Comment (manstis) [reply manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java:123]: @wtrocki @danielezonca @robpblake @kelvah FYI.

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/BaseResponse.java [hunk 4]
```diff
@@ -2,8 +2,10 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(name = "ObjectReference")
```
Comment (danielezonca) [inline infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/BaseResponse.java:8]: Is it a convention? Should the "root" object be named as `ObjectReference` or are there other reasons?
Btw can you please create a `OpenAPIConstants` class in the `infra` module so that we centralise similar strings?

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/PagedListResponse.java [hunk 5]
```diff
@@ -6,8 +6,11 @@
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.redhat.service.smartevents.infra.models.ListResult;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(name = "List",
```
Comment (danielezonca) [inline infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/PagedListResponse.java:12]: Sort of the same question here

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java [hunk 6]
```diff
@@ -6,8 +6,11 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.redhat.service.smartevents.infra.api.APIConstants;
 import com.redhat.service.smartevents.infra.exceptions.BridgeError;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(name = "Error",
```
Comment (danielezonca) [inline infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java:12]: And here

**Comment by danielezonca** [comment]
@jomarko 
If now RHOSE API are valid please keep the GHA job enable :)

Well done 👍

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/BaseResponse.java (hunk 4 above)
Comment (wtrocki) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/BaseResponse.java:8]: Sadly that is what we got from AppSRE and every project replicates it:

https://github.com/bf2fc6cc711aee1a0c2a/srs-fleet-manager/blob/10f80db6735befba95ded8ba8c61bad824501206/core/src/main/java/org/bf2/srs/fleetmanager/rest/publicapi/beans/ObjectReference.java

File: openapi/rhose-api.yaml [hunk 7]
```diff
@@ -877,6 +974,16 @@ components:
           type: string
         enabled:
           type: boolean
+    Error:
+      allOf:
+      - $ref: '#/components/schemas/ObjectReference'
+      - $ref: '#/components/schemas/Error'
```
Comment (wtrocki) [inline openapi/rhose-api.yaml:980]: This looks strange. Error references itself? Might need fix

File: openapi/rhose-api.yaml [hunk 8]
```diff
@@ -895,30 +1002,38 @@ components:
...
-        code:
-          type: string
-        reason:
-          type: string
-    ErrorsResponse:
+        items:
+          type: array
+          items:
+            $ref: '#/components/schemas/Error'
+    List:
```
Comment (wtrocki) [inline openapi/rhose-api.yaml:1014]: I do not see us actually using list anywhere? The point is to return List for every non plural response. 
We require it only because that will give us out of the box migration to KCP API thru SDK generator.

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java (hunk 3 above)
Comment (jomarko) [reply manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java:123]: @manstis hello, is there please a clear answer, which API methods return `ErrorResponse` and hich API methods return `ErrorsResponse`. I am asking to know, if I should:
- use both `content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = ErrorResponse.class))` and `content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = ErrorsResponse.class))`
- remove `content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = ErrorResponse.class))` completely as we do not know what object is returned at compile time

@wtrocki do you mean disabling the rule on our site? in `.github/workflows/openapi_validate.yaml`?

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/PagedListResponse.java (hunk 5 above)
Comment (jomarko) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/PagedListResponse.java:12]: @wtrocki am I correct here, for the `List`, is the answer same as for `ObjectReference`?

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java (hunk 6 above)
Comment (jomarko) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java:12]: @wtrocki am I correct here, for the `Error`, is the answer same as for `ObjectReference`?

File: openapi/rhose-api.yaml (hunk 7 above)
Comment (jomarko) [reply openapi/rhose-api.yaml]: fixed as https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/861/commits/49441d9e380db482ed116c27f5a0eba54c198e1d

File: openapi/rhose-api.yaml (hunk 8 above)
Comment (jomarko) [reply openapi/rhose-api.yaml:1013]: This `List` openapi component is a consequence of my changes in `PagedListResponse.java`, especially:
```java
@Schema(name = "List",
        allOf = { PagedListResponse.class, ListResponse.class })
public abstract class PagedListResponse<T> extends ListResponse<T> {
```

The issue is, without the changes above, the `spectral lint` check reports this error:
```
  821:11    error  rhoas-list-schema              Missing required schema object "List"                                                           components.schemas
```

@wtrocki so what is your advice please?

Comment (jomarko) [reply openapi/rhose-api.yaml:1013]: see my changes in https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/861/commits/f267fec87ce979bcaba4280bdf61827f433ad453

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java (hunk 3 above)
Comment (wtrocki) [reply manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java:123]: yes

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/PagedListResponse.java (hunk 5 above)
Comment (wtrocki) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/PagedListResponse.java:13]: yes

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java (hunk 6 above)
Comment (wtrocki) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java:13]: yes

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java (hunk 3 above)
Comment (manstis) [reply manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java:123]: tl; dr; All RHOSE API methods should return `ErrorsResponse`; affecting the whole backend and UX codebase.

@jomarko ALL of our API methods should be reviewed.

The `POST` and `PUT` methods (`createXXX` type calls) definately have the potential to return `ErrorsResponse` as a request to create something can contain multiple (validation) errors. Their current implementation return EITHER `ErrorResponse` or `ErrorsResponse`. They should probably ALWAYS return `ErrorsResponse` with a single item if there is only one error.

The `GET` methods also have the potential to return multiple Errors (since we use Java Bean Validation to validate the page number and page size). A `GET` with values `-1` for these parameters will return two validation errors.

@wtrocki @danielezonca @robpblake This blocks merge of this PR really.

Comment (wtrocki) [reply manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java:123]: > This blocks merge of this PR really.

+1 to change everywhere to ErrorsList if you need it.

Comment (jomarko) [reply manager/src/main/java/com/redhat/service/smartevents/manager/api/user/ProcessorsAPI.java]: @manstis I tried to change all API methods to return `ErrorsResponse` instead of `ErrorResponse`. My understadning is that changes are needed at two places:
- Annotations on 'API Interface'
- In real backend implementation

I tend to think the backend is already fine in:
- `ConstraintViolationExceptionMapper`

However I am not sure about my changes in:
- `ExternalUserExceptionMapper`
- `InternalPlatformExceptionMapper`

Would you mind to review my latest commit?

**Comment by manstis** [comment]
Hi @jomarko my [PR](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/894) has been merged so you can update this now :+1:

**Comment by jomarko** [comment]
@wtrocki hello, would you mind to review latest changes please?

**Comment by jomarko** [comment]
@dimakis identified, that inheritance is causing issues for spectral lint. He should get back with proposed next steps.

**Review by dimakis — COMMENTED** [review]
A follow up is that, most other API endpoints follow the `/api/project_name/v1/blah/{id}/blah` instead of the proposed `/api/v1/project_namev/blah/{id}/blah`.

sorry about this but I've just got word that we should revert them all to the former as it there would be too many breaking changes otherwise.

**Comment by manstis** [comment]
> A follow up is that, most other API endpoints follow the `/api/project_name/v1/blah/{id}/blah` instead of the proposed `/api/v1/project_namev/blah/{id}/blah`.
> 
> sorry about this but I've just got word that we should revert them all to the former as it there would be too many breaking changes otherwise.

@dimakis We've just completed our move to `/api/project_name/v1/blah...` are you saying we now need to roll it back?

@robpblake @danielezonca FYI.

**Comment by wtrocki** [comment]
@manstis That was issue with "app interface guidelines" and real implementations that kafka and connectors took. 
No one followed them so we went and decided to adjust it.  Really sorry for making this problematic.
URLs should be hidden from users by SDKs and we could make this by search replace

**Comment by wtrocki** [comment]
> We've just completed our move to /api/project_name/v1/blah..

Apologies for confusion. If you have moved to /api/smartevents_mgmt/v1/bridges etc. then we do not have issue.

**Comment by jomarko** [comment]
@dimakis @dimakis I have pushed changes you asked for, would you mind to re-review?

File: .github/workflows/openapi_validate.yaml [hunk 9]
```diff
@@ -16,4 +14,8 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: 15.x
-      - run: npx @stoplight/spectral@5.9.0 lint ./openapi/smartevents_mgmt.yaml
\ No newline at end of file
+####### rhoas-list-schema skipped due to https://issues.redhat.com/browse/MGDOBR-935
+      - run: |
+          npx @stoplight/spectral@5.9.0 lint
+          ./openapi/smartevents_mgmt.yaml \
+          --skip-rule rhoas-list-schema
```
[the remark this answers is not in the document]
Comment (jomarko) [reply .github/workflows/openapi_validate.yaml]: @dimakis hello, there was a missing backslash, the `npx` works now

[the remark this answers is not in the document]
Comment (wtrocki) [reply .github/workflows/openapi_validate.yaml]: Thank you @jomarko 
+1 to keep things simple. @dimakis we need to improve running rules in future release.

File: manager/src/main/resources/application.properties [hunk 10]
```diff
@@ -13,9 +13,10 @@ quarkus.smallrye-openapi.info-title=Red Hat Openshift SmartEvents Fleet Manager
 quarkus.smallrye-openapi.info-version=0.0.1
 quarkus.smallrye-openapi.info-description=The API exposed by the fleet manager of the SmartEvents service.
 quarkus.smallrye-openapi.info-license-name=Apache 2.0
-quarkus.smallrye-openapi.info-license-url=https://www.apache.org/licenses/LICENSE-2.0.html
+quarkus.smallrye-openapi.info-license-url=https://www.apache.org/licenses/LICENSE-2.0
 quarkus.smallrye-openapi.info-contact-email=openbridge-dev@redhat.com
 quarkus.smallrye-openapi.info-contact-name=Development Team
+quarkus.smallrye-openapi.servers=https://api.openshift.com,https://api.stage.openshift.com,http://localhost:8000,/
```
Comment (manstis) [inline manager/src/main/resources/application.properties:19]: Should this really include the trailing comma (`http://localhost:8000,`)?

**Review by manstis — APPROVED** [review]
IDK if we _leak_ something that is not ideal; but there is probably little we can do about it.

https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/1d085f5855250e8aa36b24bff85555019814629e/openapi/smartevents_mgmt.yaml#L1162

I'm also not a fan of renaming some of the classes for the OpenAPI file (e.g. `ErrorsResponse` -> `ErrorsList` and `PagedListResponse` -> `List` especially given we still keep `ListResponse`) as IMO they make their naming inconsistent with everything else we have and in some places more confusing. This is an opinion and nothing technical and so is not blocking the PR.

Otherwise a minor comment and based on the other expert reviews LGTM :+1:

File: manager/src/main/resources/application.properties (hunk 10 above)
Comment (jomarko) [reply manager/src/main/resources/application.properties:19]: it is not trailing comma, it is comma before last server url, if I remove `/` form the list, I will get:
```
   14:3   warning  rhoas-servers-config           OpenAPI `servers` object is missing the following required URLs: /                              servers[0]
```

Comment (manstis) [reply manager/src/main/resources/application.properties:19]: Thank-you for pointing it out :smile:

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java [hunk 11]
```diff
@@ -2,14 +2,26 @@
...
+        allOf = { BaseResponse.class })
 public class ErrorResponse extends BaseResponse {
 
+    // https://issues.redhat.com/browse/MGDOBR-935
+    @JsonProperty("id")
+    private String id;
+
+    // https://issues.redhat.com/browse/MGDOBR-935
+    @JsonProperty("href")
+    private String href;
```
Comment (danielezonca) [inline infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java:23]: As far as I see [`BaseResponse`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/BaseResponse.java#L9) also has `kind` and `name` as fields. If I understand the bug correctly you are duplicating fields from parent class as workaround. But why not also the other two fields?

**Review by danielezonca — APPROVED** [review]
LGTM, two minor comments

File: manager/src/main/resources/application.properties (hunk 10 above)
[the remark this answers is not in the document]
Comment (jomarko) [reply manager/src/main/resources/application.properties:19]: sent email to quarkus-dev, lets see

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java (hunk 11 above)
Comment (jomarko) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java:23]: good question, I think it is just due to the validation bug. If we have a look on the openapi:
```
    Error:
      type: object
      allOf:
      - $ref: '#/components/schemas/ObjectReference'
      properties:
        id:
          type: string
        href:
          type: string
        code:
          type: string
        reason:
          type: string
```

We can seen there `allOf` referring to the parent `ObjectReference`. So I believe the SDK will be generated correctly, I mean `Error` model object for the SDK.

@dimakis @wtrocki could you please confirm my thoughts?

File: manager/src/main/resources/application.properties (hunk 10 above)
[the remark this answers is not in the document]
Comment (jomarko) [reply manager/src/main/resources/application.properties:19]: yes, see https://issues.redhat.com/browse/MGDOBR-938

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java (hunk 11 above)
Comment (dimakis) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java:23]: yes @danielezonca, I believe you are correct with your thoughts here. @wtrocki may have some more insight but from my understanding that is accurate

Comment (wtrocki) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java:23]: Generators require all properties to be part of the allOff. Mike did fix the same problem in this PR: 
https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/pull/204/files#diff-28a7ff5a0ccb4c306af5199105457eb5c1547945322341cdcac828b585e09f5c

Reapplying solution from that PR would give us much nicer result schema (and generally set good pattern for object inheritance in the future)

Comment (MikeEdgar) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/models/responses/ErrorResponse.java:23]: What I saw in the openapi client generator earlier is that the `allOf` with only the parent class results in Java code properly having `Bean extends Parent` definitions. However, it results in invalid JavaScript clients. 

Using `allOf` with _both_ parent and child, the resulting Java puts everything into a single class (not actual inheritance at the Java level), but results in JavaScript having constructs like `export type ConsumerGroup = ConsumerGroupAllOf & ObjectReference;`. Personally I find the naming of the `ConsumerGroupAllOf` object confusing since it only includes the first `allOf` entry's properties.

Bottom line is that I think there is a lot of room for improvement in the code generator's handling of this both with consistency between languages and creating less "stuff" in the output. The Java output, for example, generates several "AllOf" classes that are not referenced elsewhere.

**Comment by jomarko** [comment]
Fails as:
```
Error:  [ERROR] Tests run: 13, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1,068.694 s <<< FAILURE! - in RunCucumberTest
Error:  [ERROR] Kafka Action tests - Send message to Kafka topic using the Kafka action  Time elapsed: 0.668 s  <<< ERROR!
java.lang.RuntimeException: Property managed.kafka.sso.auth-server-url was not defined.
	at com.redhat.service.smartevents.integration.tests.common.Utils.getSystemProperty(Utils.java:9)
	at com.redhat.service.smartevents.integration.tests.resources.kafka.KafkaConnectionParameters.getInstance(KafkaConnectionParameters.java:30)
	at com.redhat.service.smartevents.integration.tests.resources.kafka.KafkaResource.createKafkaTopic(KafkaResource.java:42)
	at com.redhat.service.smartevents.integration.tests.steps.KafkaSteps.createKafkaTopic(KafkaSteps.java:25)
	at ✽.create a new Kafka topic "topic1"(classpath:features/action-kafka.feature:8)
```

**Comment by wtrocki** [comment]
Thank you so much @jomarko
